commons/src/main/scala/dagr/commons/reflect/ReflectiveBuilder.scala

```diff
@@ -31,8 +54,7 @@ class ReflectiveBuilder[T](val clazz: Class[T]) {
   extractParameterInformation(constructor)
 
   /**
-    * Finds the appropriate constructor to reflect on. Either the single constructor, or when there are multiple
-    * the one annotated with @CLPConstructor, or failing that the first from the list.
+    * Finds the appropriate constructor to reflect on. The constructor must be public.
     */
   private def findConstructor : MethodSymbol = pickConstructor {
     typeSignature.members.toSeq.filter(p => p.isConstructor && p.owner == clazzSymbol && p.isPublic).map(_.asMethod)
```

Comment:
Commenting here because github won't allow me to comment below.  I think pickConstructor should be changed to look two ways now:
1) if there's a single ctor return it
2) if there's more than one, but one of the returns true from `ctor.isPrimaryConstructor` return that one
3) Else exception